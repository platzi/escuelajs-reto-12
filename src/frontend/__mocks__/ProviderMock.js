import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import reducer from '../reducers';

const initialState = {
  total_price: 0,
  cart: [],
  products: [
    {
      id: '1',
      image: 'https://arepa.s3.amazonaws.com/camiseta.png',
      title: 'Camiseta',
      price: 25,
      description: 'bla bla bla bla bla',
    },
    {
      id: '3',
      image: 'https://arepa.s3.amazonaws.com/mug.png',
      title: 'Mug',
      price: 10,
      description: 'bla bla bla bla bla',
    },
    {
      id: '4',
      image: 'https://arepa.s3.amazonaws.com/pin.png',
      title: 'Pin',
      price: 4,
      description: 'bla bla bla bla bla',
    },
    {
      id: '5',
      image: 'https://arepa.s3.amazonaws.com/stickers1.png',
      title: 'Stickers',
      price: 2,
      description: 'bla bla bla bla bla',
    },
    {
      id: '6',
      image: 'https://arepa.s3.amazonaws.com/stickers2.png',
      title: 'Stickers',
      price: 2,
      description: 'bla bla bla bla bla',
    },
    {
      id: '7',
      image: 'https://arepa.s3.amazonaws.com/hoodie.png',
      title: 'Hoodie',
      price: 35,
      description: 'bla bla bla bla bla',
    },
  ],
};

const store = createStore(
  reducer,
  initialState,
  compose(applyMiddleware(thunk)),
);
const history = createBrowserHistory();

const ProviderMock = props => (
  <Provider store={store}>
    <Router history={history}>{props.children}</Router>
  </Provider>
);

export default ProviderMock;
