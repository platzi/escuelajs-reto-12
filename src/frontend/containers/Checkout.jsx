/* eslint-disable react/no-array-index-key */
import React from 'react';
import { connect } from 'react-redux';
import '../styles/components/Checkout.styl';
import { removeToCart } from '../actions';

const Checkout = (props) => {
  const { cart, totalPrice } = props;

  const handleRemoveToCart = (item, key) => {
    props.removeToCart(item, key);
  };

  return (
    <div className="Checkout">
      <div className="Checkout-content">
        {cart.length > 0 ? <h3>Lista de Pedidos:</h3> : <h2>Sin Pedidos</h2>}
        {cart.map((item, key) => (
          <div className="Checkout-item" key={key}>
            <div className="Checkout-element">
              <h4>{item.title}</h4>
              <span>
$
                {item.price}
              </span>
            </div>
            <i
              role="button"
              tabIndex={-1}
              className="fas fa-trash-alt"
              onClick={() => handleRemoveToCart(item, key)}
            />
          </div>
        ))}
      </div>
      {cart.length > 0 && (
        <div className="Checkout-sidebar">
          <h3>Precio Total:</h3>
          <h4>
$
            {totalPrice}
          </h4>
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    cart: state.cart,
    totalPrice: state.total_price,
  };
};

const mapDisptachToProps = {
  removeToCart,
};

export default connect(
  mapStateToProps,
  mapDisptachToProps,
)(Checkout);
