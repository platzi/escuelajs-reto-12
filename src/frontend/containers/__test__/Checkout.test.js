import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Checkout from '../Checkout';

describe('Checkout component', () => {
  test('Match Snapshot', () => {
    const checkout = create(
      <ProviderMock>
        <Checkout />
      </ProviderMock>,
    );
    expect(checkout.toJSON()).toMatchSnapshot();
  });
});
