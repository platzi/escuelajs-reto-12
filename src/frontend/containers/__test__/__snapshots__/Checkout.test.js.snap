// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component Match Snapshot 1`] = `
<div
  className="Checkout"
>
  <div
    className="Checkout-content"
  >
    <h2>
      Sin Pedidos
    </h2>
  </div>
</div>
`;
